background/addon-state.js

```diff
@@ -29,6 +30,27 @@ class AddonState {
     };
 
     browser.runtime.onMessage.addListener(async (message, {tab}) => {
+      if (message.command === "enableContentScript") {
+        this.enableContentScript(message.value);
+      } else {
+        this.handleColorMessage(message, {tab});
+      }
+    });
+
+    this.enableContentScript = async (value) => {
+      if (value === true) {
+        let file = "data/contentscript.js";
+        this.state.contentScriptObj = await browser.contentScripts.register({
+          matches: ["<all_urls>"],
+          js: [{file}],
+        });
+      } else if (this.state.contentScriptObj !== null) {
+        await this.state.contentScriptObj.unregister();
+        this.state.contentScriptObj = null;
+      }
+    };
+
+    this.handleColorMessage = async (message, {tab}) => {
```

Comment:
You don't really need all this (and all the places you send the message), `refreshAddon` will do it for you since it's called on every setting changed.

Reply:
Oh yeah you are right, this is way easiert. Changed it, so the whole registering stuff is in the onInit() function